conf/route_control.py

```diff
@@ -667,7 +667,7 @@ def _parse_route_entry_msg(self, route_entry: dict) -> Optional[RouteEntry]:
         if not attr_dict.get(KEY_INTERFACE):
             return None
         interface_index = int(attr_dict.get(KEY_INTERFACE))
-        interface = self._ipdb.interfaces[interface_index].ifname
+        interface = self._ndb.interfaces[interface_index]["ifname"]
```

Comment:
I would encourage using `.get()` and returning when the output is none. It looks like we have risks of having key errors here by specifying keys of nested dicts. Do we know for sure that those keys will be there?

Reply:
This comment applies at a couple of places

Reply:
> I would encourage using `.get()` and returning when the output is none. It looks like we have risks of having key errors here by specifying keys of nested dicts. Do we know for sure that those keys will be there?

Sure, I am going to change to `.get()`.

Regarding the `keys`, yes, they will be there for sure. Here are the keys that are part of the `interfaces` table:
```shell
$ python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pyroute2 import NDB
>>> ndb = NDB()
>>> print(ndb.interfaces[1])
{'state': 'up', 'target': 'localhost', 'index': 1, 'tflags': 0, 'family': 0, 'ifi_type': 772, 'flags': 65609, 'change': 0, 'address': '00:00:00:00:00:00', 'broadcast': '00:00:00:00:00:00', 'ifname': 'lo', 'mtu': 65536, 'link': None, 'qdisc': 'noqueue', 'master': None, 'txqlen': 1000, 'operstate': 'UNKNOWN', 'linkmode': 0, 'net_ns_pid': None, 'ifalias': None, 'num_vf': None, 'group': 0, 'ext_mask': None, 'promiscuity': 0, 'num_tx_queues': 1, 'num_rx_queues': 1, 'carrier': 1, 'carrier_changes': 0, 'link_netnsid': None, 'phys_port_name': None, 'proto_down': 0, 'gso_max_segs': 65535, 'gso_max_size': 65536, 'event': None, 'new_netnsid': None, 'if_netnsid': None, 'carrier_up_count': 0, 'carrier_down_count': 0, 'new_ifindex': None, 'min_mtu': 0, 'max_mtu': 0, 'alt_ifname': None, 'parent_dev_name': None, 'parent_dev_bus_name': None, 'gro_max_size': None, 'tso_max_size': None, 'tso_max_segs': None, 'kind': None, 'slave_kind': None, 'xdp_attached': None, 'xdp_flags': None, 'xdp_prog_id': None, 'xdp_drv_prog_id': None, 'xdp_skb_prog_id': None, 'xdp_hw_prog_id': None}
```

Reply:
Done, changed to use `.get()`